### ISSUE TITLE
[Question] Styling input fields when focused?

### ISSUE THREAD

**Issue by alanbuchanan** [gcanti/tcomb-form-native#466]
This question has been asked before, but no answer was pointed out. I can't find anything specific in the readme or the stylesheet guide.

Can someone please give an example of changing the style of the input field (for example changing border color) when the field is focused?

**Comment by Dreyer** [comment]
What is the issue ID of the other question? I'm thinking this is a common use case that *should* be simpler. There should be a way to style for `hasFocus` in the Stylesheet similar to `hasError` in the templates.

The only way I could achieve this was by supplying a custom template **and** factory option for the `name` field as an example:

<img src="https://user-images.githubusercontent.com/294523/37588404-0d888dae-2b5a-11e8-9048-c98db3f008e6.png" height="300" border="1" />

https://snack.expo.io/HJ-bIbpKf

As far as I can see, the major downside is that you would have to do this for every field that needed this `hasFocus` functionality and that might mean reimplementing a template and factory for every supported type of component. 

Alternatively, if there is a way to override the actual `Textbox` or [`Component`](https://github.com/gcanti/tcomb-form-native/blob/master/lib/components.js#L59) component itself so that all components would inherit this stylesheet functionality. Would probably also mean changing the `lib/stylesheets/bootstrap.js` to include `focus` (alongside `normal`, `error`, `notEditable`).

**Comment by nederev** [comment]
agree, this should be natively supported with hasFocus  stylesheet options like normal/error

**Comment by nederev** [comment]
This should be much simpler than that :( Such a wonderful library but without onFocus styling support its not complete.

Really ugly hack, change border color on field focus:

```
  const stylesheet = _.cloneDeep(t.form.Form.stylesheet);

  formOptions = {
    stylesheet: stylesheet,
    fields: {
      name: {
        stylesheet: _.cloneDeep(stylesheet),
        onFocus: () => this.changeBorderColorOf('name', true),
        onBlur: () => this.changeBorderColorOf('name', false),
      },
      phoneNumber: {
        stylesheet: _.cloneDeep(stylesheet),
        onFocus: () => this.changeBorderColorOf('phoneNumber', true),
        onBlur: () => this.changeBorderColorOf('phoneNumber', false),
      },
    }
  }


  changeBorderColorOf(fieldName, isFocused) {
    const copyFormOptions = _.cloneDeep(this.formOptions);
    copyFormOptions.fields[fieldName].stylesheet.textbox.normal.borderColor = isFocused ? '#000' : 'rgba(0, 0, 0, 0.2)';
    this.setState({
      formOptions: copyFormOptions
    });
  }

  onChangeText = (formValues: Person) => {
    this.setState({
      formValues,
    });
  }
   <Form
      value={this.state.formValues}
      type={Person}
      options={this.state.formOptions}
      onChange={this.onChangeText}
   />

```
